pkg/oci/load_balancer_spec.go

```diff
@@ -244,32 +256,50 @@ func getBackends(nodes []*v1.Node, nodePort int32) []loadbalancer.BackendDetails
 	return backends
 }
 
-func getBackendSets(svc *v1.Service, nodes []*v1.Node) map[string]loadbalancer.BackendSetDetails {
+func getBackendSets(svc *v1.Service, nodes []*v1.Node, sslCfg *SSLConfig) map[string]loadbalancer.BackendSetDetails {
 	backendSets := make(map[string]loadbalancer.BackendSetDetails)
 	for _, servicePort := range svc.Spec.Ports {
 		name := getBackendSetName(string(servicePort.Protocol), int(servicePort.Port))
-		backendSets[name] = loadbalancer.BackendSetDetails{
-			Policy:        common.String(DefaultLoadBalancerPolicy),
-			Backends:      getBackends(nodes, servicePort.NodePort),
-			HealthChecker: getHealthChecker(svc),
+		port := int(servicePort.Port)
+		sslConfig := getSSLConfiguration(sslCfg, port)
+		if sslConfig != nil {
+			backendSets[name] = loadbalancer.BackendSetDetails{
+				Policy:           common.String(DefaultLoadBalancerPolicy),
+				Backends:         getBackends(nodes, servicePort.NodePort),
+				HealthChecker:    getHealthChecker(sslCfg, port, svc),
+				SslConfiguration: sslConfig,
+			}
+		} else {
+			backendSets[name] = loadbalancer.BackendSetDetails{
+				Policy:        common.String(DefaultLoadBalancerPolicy),
+				Backends:      getBackends(nodes, servicePort.NodePort),
+				HealthChecker: getHealthChecker(sslCfg, port, svc),
+			}
 		}
-
 	}
 	return backendSets
 }
 
-func getHealthChecker(svc *v1.Service) *loadbalancer.HealthCheckerDetails {
-	path, port := apiservice.GetServiceHealthCheckPathPort(svc)
-	if path != "" {
+func getHealthChecker(cfg *SSLConfig, port int, svc *v1.Service) *loadbalancer.HealthCheckerDetails {
+	var protocol string
```

Comment:
```go
// If the health-check has SSL enabled use TCP rather than HTTP.
protocol := lbNodesHealthCheckProtoHTTP
if cfg != nil && cfg.Ports.Has(port) {
	protocol = lbNodesHealthCheckProtoTCP
}		
```